gemma.cc

```diff
@@ -602,21 +618,22 @@ void GenerateImpl(GemmaImpl<TConfig>& gemma, size_t max_tokens,
   size_t pos_offset = 0;  // offset relative to pos
   const double prefill_start = hwy::platform::Now();
 
-  // Prefill stops before prompt.size() - 1 since the last prompt token is the
+  // Prefill stops before prompt_size - 1 since the last prompt token is the
   // first input token for generation.
-  while (pos_offset < prompt.size() - 1) {
-    const size_t end_offset =
-        std::min(kPrefillBatchSize, prompt.size() - 1 - pos_offset);
-    HWY_DASSERT(end_offset < prompt.size());
+  while (pos_offset < prompt_size - 1) {
+    const size_t batch_size =
+        std::min(kPrefillBatchSize, prompt_size - 1 - pos_offset);
+    HWY_DASSERT(batch_size <= kPrefillBatchSize);
+    HWY_DASSERT(pos_offset + batch_size <= prompt_size - 1);
     const int* batch_tokens = prompt.data() + pos_offset;
-    Prefill<TConfig, kPrefillBatchSize>(batch_tokens, end_offset, pos,
+    Prefill<TConfig, kPrefillBatchSize>(batch_tokens, batch_size, pos,
```

Comment:
I think the original reason we went with end_offset is that the common conception of a "batch size" is typically reserved for a batch of observations/examples, rather than multiple tokens computed in parallel. Granted maybe we should have been consistent and gotten rid of other uses of the "batch terminology".

For now I'm fine w/ this just be aware it may be confusing. If we ever add support for bulk offline inference (either in the core code or as a frontend feature/application example) in the future, then it will be really confusing and we should probably change it then.

Reply:
ACK :)